.changeset/silly-clocks-tan.md

```diff
@@ -0,0 +1,63 @@
+---
+"@fluidframework/tree": major
+---
+
+Implement compatibility-based schema evolution API
+
+This change adjusts some top-level APIs for using SharedTree in a context that may necessitate changing a document's schema.
+The motivation for these changes is to allow future relaxation of `SharedTree`'s restrictions around view schema and stored schema compatibility,
+which will enable more flexible policies around how applications can update their documents' schemas over time.
+
+Application authors are encouraged to develop a compatibility policy which they are comfortable with using the guidance in the
+"Schema Evolvability" section of `@fluidframework/tree`'s readme.
+
+To make the details of schema compatibilities that SharedTree supports more clear,
+`TreeView.error` has been functionally replaced with the `compatibility` property.
+Users desiring the previous strict behavior should use `view.compatibility.isEquivalent` at appropriate places in application logic.
+
+# `ITree.schematize` deprecation
+
+`ITree.schematize` has been deprecated in favor of `ITree.viewWith`.
+Unlike `schematize`, `viewWith` does not implicitly initialize the document. As such, it doesn't take an `initialTree` property.
+Instead, applications should initialize their trees in document creation codepaths using the added `TreeView.initialize` API.
+
+As an example, something like the following code may have been used before for both the document create and document load codepaths:
+
+```typescript
+const tree = SharedTree.create(runtime, "foo");
```

Comment:
Use of the encapsulated API which is not public seems non-ideal (at least as the only example).

It might be nice to include an example of how we expect users of container schema to use the new API. (maybe both the static/in schema case and also the dynamic DDS creation case)